commands/core/init.drush.inc

```diff
@@ -0,0 +1,115 @@
+  );
+  return $items;
+}
+
+/**
+ * Initialize local Drush configuration
+ */
+function drush_init_core_init() {
+  $home = drush_server_home();
+  $drush_config_dir = $home . "/.drush";
+  $drush_config_file = $drush_config_dir . "/drushrc.php";
+  $drush_bashrc = $drush_config_dir . "/drush.bashrc";
+  $examples_dir = DRUSH_BASE_PATH . "/examples";
+  $example_configuration = $examples_dir . "/example.drushrc.php";
+  $example_bashrc = $examples_dir . "/example.bashrc";
+  $bashrc_additions = array();
+
+  // Create a ~/.drush directory if it does not yet exist
+  if (!is_dir($drush_config_dir)) {
+    drush_mkdir($drush_config_dir);
+  }
+
+  // If there is no ~/.drush/drushrc.php, then copy the
+  // example Drush configuration file here
+  if (!is_file($drush_config_file)) {
+    @copy($example_configuration, $drush_config_file);
+    drush_log(dt("Copied example Drush configuration file to !path", array('!path' => $drush_config_file)), 'ok');
+  }
+
+  // If there is no ~/.drush/drush.bashrc file, then copy
+  // the example bashrc file here
+  if (!is_file($drush_bashrc)) {
+    @copy($example_bashrc, $drush_bashrc);
+    $bashrc_additions = array("%$drush_bashrc%" => "# Include Drush bash customizations\n. $drush_bashrc\n\n");
+    drush_log(dt("Copied example Drush bash configuration file to !path", array('!path' => $drush_bashrc)), 'ok');
+  }
+
+  // Decide whether we want to add our Bash commands to
+  // ~/.bashrc or ~/.bash_profile
+  $bashrc = drush_init_find_bashrc($home);
+
+  // If drush is not in the $PATH, then figure out which
+  // path to add so that Drush can be found globally.
+  if (!drush_which("drush")) {
+    $drush_path = drush_find_path_to_drush();
+    $drush_path = preg_replace("%^$home/%", '$HOME/', $drush_path);
+
+    $bashrc_additions = array("%$drush_path%" => "# Path to Drush, added by 'drush init'\n\$PATH=\$PATH:$drush_path\n\n");
+  }
+
+  // Modify the users bashrc file, adding our customizations.
+  $bashrc_contents = "";
+  if (file_exists($bashrc)) {
+    $bashrc_contents = file_get_contents($bashrc);
+  }
+  $new_bashrc_contents = $bashrc_contents;
+  foreach ($bashrc_additions as $pattern => $addition) {
+    // Only put in the addition if the pattern does not already
+    // exist in the bashrc file.
+    if (!preg_match($pattern, $new_bashrc_contents)) {
+      $new_bashrc_contents = $addition . $new_bashrc_contents;
+    }
+  }
+  if ($new_bashrc_contents != $bashrc_contents) {
+    file_put_contents($bashrc, $new_bashrc_contents);
+    drush_log(dt("Updated bash configuration file !path", array('!path' => $bashrc)), 'ok');
+  }
+}
+
+/**
+ * Determine which .bashrc file is best to use on this platform.
+ */
+function drush_init_find_bashrc($home) {
+  return $home . "/.bashrc";
+}
+
+/**
+ * Determine where Drush is located, so that we can add
+ * that location to the $PATH
+ */
+function drush_find_path_to_drush($home) {
+  // First test: is Drush inside a vendor directory?
```

Comment:
Yeah, I'd go with selecting site-local Drush as well. This does mean that wiping a random codebase can unknowingly ruin your drush command. If would be good if init were idempotent so that folks could rerun in a different codebase and get up to speed. Is it idempotent already? I'm OK with extra junk getting appended to bash files in this unusual case.